### ISSUE TITLE
[Bug]: Infinite Loop in WebDriver Initialization Prevents Browser Navigation

### ISSUE THREAD

**Issue by mahmoudali01** [ShaftHQ/SHAFT_ENGINE#1514]
**Describe the bug**

Trying to pass the Chrome option that opens an existing cached Chrome profile, makes the code stuck in an infinite loop of initializing the Chrome driver 

**Environment**

 1. SHAFT_Engine version that exhibits the issue (Note: We support only the latest release): 8.1.20240224
 2. Last SHAFT_Engine version that did not exhibit the issue (if applicable):
 3. Desktop OS: Windows
 4. Desktop Browser (if applicable): Chrome
 5. Desktop Browser Version (if applicable): 121.0.6167.185

**To Reproduce**

1. initialize an instance of Chrome options 
2. Add the following arguments "user-data-dir", "profile-directory"
3. Pass the chrome options to the WebDriver and Set the DriverType to chrome
4. Navigate to any website

**Expected behavior**

Chrome driver should be opened and then navigate to the specified website

**Screenshots**

![Screenshot_4](https://github.com/ShaftHQ/SHAFT_ENGINE/assets/49574153/8b5797a6-58bf-4044-adea-5746ee8d97f1)

**Additional context**

Sample of the test class:

package test;

import com.shaft.driver.DriverFactory;
import com.shaft.driver.SHAFT;
import org.openqa.selenium.chrome.ChromeOptions;
import org.testng.annotations.Test;

public class main {
    public static SHAFT.GUI.WebDriver driver;

    @Test
    public void openChrome(){
        String userDataDirectory = "C:/Users/User/AppData/Local/Google/Chrome/User Data";
        String profileDirectory = "Profile 2";
        ChromeOptions chromeOptions = new ChromeOptions();
        chromeOptions.addArguments("user-data-dir=" + userDataDirectory);
        chromeOptions.addArguments("profile-directory=" + profileDirectory);
        driver = new SHAFT.GUI.WebDriver(DriverFactory.DriverType.CHROME,chromeOptions);
        driver.browser().navigateToURL("https://www.google.com");
    }
}

**Comment by mahmoudali01** [comment]
***update***
I added a new line in the code `Runtime.getRuntime().exec("taskkill /f /im chrome.exe");` to remove any active Chrome session and  still initiates the driver multiple times but what changed it's not stuck on an infinite loop it opens and closes the Chrome driver multiple times and then throws this error:
`java.lang.AssertionError: Driver Factory Action "createNewLocalDriverInstance" failed. With the following test data "Failed to create new Browser Session". Root cause: "org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: unknown error: Failed to initialize BiDi Mapper: TypeError: Failed to set the 'innerHTML' property on 'Element': This document requires 'TrustedHTML' assignment."
`

**Comment by mahmoudali01** [comment]
***update***
it works fine now after adding this property attribute `enableBiDi=false` in src/main/resources/properties/ExecutionPlatform.properties file

**Comment by MohabMohie** [comment]
thanks for letting us know mahmoud. I'm glad that you worked around the issue.
I've identified the root cause of the seemingly infinite loop (it is actually hardwired to do 6 attempts but it may have seemed like an infinite loop because it took so much time)
As of the new release if anyone faces this exception then SHAFT will fail immediately with a proper exception message.
I've also added a fix for the "Failed to initialize BiDi Mapper" which will automatically disable BiDi and try again.
